### PR TITLE
Train 159 (targeting 1.8.9)

### PR DESCRIPTION
**Pull request by lingmann**
## High Level Description

Train 159 targeting 1.8.9 branch

#1494 Backporting https://jira.mesosphere.com/browse/DCOS-10327 to 1.8.9
#1458 Update cosmos to version 0.2.1
#1456 Backporting the fix for DCOS_OSS-683 to 1.8
#1445 Bumped Mesos to include fixes for leakage of sensitive data - Mesos 1.0.4

### CODE REVIEW

**Comment by lingmann** [comment]
Both `integration-test/deploy-vpc-cli` and `integration-test/upgrade-onprem` are failing with `AssertionError: Number of hosts must be equal to sum of masters and agents plus the bootstrap host`. The failure signature is different though in each.

* integration-test/deploy-vpc-cli shows: `host.num_masters: 1, num_agents: 2, num_public_agents: 1` followed by **2 hosts**
* integration-test/upgrade-onprem shows: `host.num_masters: 3, num_agents: 2, num_public_agents: 1` followed by **6 hosts**

I've reproduced this 2x so far.

There is a possibility this is related to https://jira.mesosphere.com/browse/DCOS-13915 cc: @orsenthil

**Comment by lingmann** [comment]
@mellenburg do you have any insight on the AssertionError above? I think there might be fixes related to this currently on master.

**Comment by mellenburg** [comment]
@lingmann Yes, that is an old issue. That has since been fixed. Specifically, we used to ask AWS to describe the hosts by getting the instances with the cloudformation tag, however, what we discovered is that the tagging process is likely asynchronous to the cloudformation claiming that the the stack is "complete". Thus, what happens is that the number of desired instances is in the stack, but the tags are not, so we only get a subset.

See: https://github.com/mesosphere/dcos/blob/train-159/test_util/aws.py#L352

The solution to this was to do a painful refactor where we use the resource model to extract the instances instead of tags, but then there is the issue that boto does not have a resource for AutoscalingGroups to cover this use case, so then this ugly function had to be invented:
https://github.com/dcos/dcos/blob/master/test_util/aws.py#L154

TL;DR
- this is a known problem that was fixed on master quite a long while ago
- the fix required for this was a pretty heavy refactor
- our test code is tied into dc/os proper so fixing this will be a giant pain as there can be no independent bump
- In an ideal world I would have split out test_util into a separate repo ages ago and now we could just bump it and be done with it.
- keep on retrying :/

**Comment by lingmann** [comment]
Ok, thank you @mellenburg for the detailed explanation! Is there any way we could have a simple patch which just disables these tests for the 1.8.x branch? It seems unlikely that we will be backporting the full fix / refactor, but it would be nice if we could still reliably run some of our tests on the 1.8.x series. WDYT?

**Comment by spahl** [comment]
@mellenburg @lingmann what is the decision on this?

**Comment by amitaekbote** [comment]
Changed the stable installer url to 1.8.8 for upgrade tests, once that passes this should be good to go

**Comment by amitaekbote** [comment]
@mesosphere-mergebot ready for review

**Comment by amitaekbote** [comment]
@spahl this is good to go for 1.8.9
